deltacat/compute/compact_driver.py

```diff
@@ -0,0 +1,58 @@
+import argparse
+import base64
+import gzip
+import json
+import logging
+import os
+import pathlib
+from typing import List
+
+import boto3
+
+import yaml
+
+from deltacat import logs
+from deltacat.autoscaler.events.dispatcher import EventDispatcher
+from deltacat.autoscaler.events.event_store import EventStoreClient
+
+from ray.autoscaler._private.aws.events import AwsEventManager
+
+from deltacat.autoscaler.events.workflow import poll_events
+
+logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
+
+COMPACTION_EVENT_SUMMARY_TABLE_NAME = "RAY_COMPACTION_EVENT_SUMMARY"
+ONE_HOUR_MS = 60 * 1000 * 60
+
+
+def partition_discovery() -> List[str]:
+    return ["PLACEHOLDER_PARTITION"]
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config_path", type=pathlib.Path, required=True)
+
+    args = parser.parse_args()
+
+    full_path = os.path.expanduser(args.config_path)
+    with open(full_path, "r") as stream:
+        config = yaml.safe_load(stream)
+
+    events_config = config["events"]
+    manager = AwsEventManager(events_config) if events_config else None
+    source_table_dict = events_config["parameters"]["sourceTable"]
+    b64_encoded_pkv = events_config["parameters"].get("partitionsToCompact")
+    if b64_encoded_pkv:
+        pkv = gzip.decompress(base64.b64decode(b64_encoded_pkv))
+        pkv_json = json.loads(pkv)
+        partition_values = [kv["value"] for kv in pkv_json[0]]
+    else:
+        logger.info(f"No partition key-values found to compact. Attempting partition discovery.")
+        partition_values = partition_discovery()
```

Comment:
Is "job runner" or "job manager" maybe a more descriptive name than "driver"? 

Also, we should try to separate this into a generic job runner (e.g. for any type of job that may run against your data catalog), a generic compaction job runner (e.g. that can run a change-data-capture job for any type of data catalog that supports deltas), and a specific compaction job runner (e.g. the implementation that we have here that has very specific expectations about the event model, payload content encoding, etc.). 

At a high-level, it seems like a generic job runner just needs to take some sort of event manager, event store, and event dispatcher as input and it should return a "handle" to the job runner as output (e.g. ideally a Ray cluster session ID, distinct cluster name, or some other ID that can be used to monitor/manage this job runner instance through the Ray CLI or another external app/service).

Reply:
These files are slightly outdated - they no longer exist in the [newer DeltaCAT fork](https://github.com/pdames/deltacat/pull/1/files) as I've moved these scripts into SungatePython.